controllers/provisioners/eksfargate/eksfargate.go

```diff
@@ -0,0 +1,290 @@
+)
+
+func New(p provisioners.ProvisionerInput) *InstanceGroupContext {
+	ctx := &InstanceGroupContext{
+		InstanceGroup: p.InstanceGroup,
+		AwsWorker:     p.AwsWorker,
+		Log:           p.Log.WithName("eks-fargate"),
+	}
+
+	instanceGroup := ctx.GetInstanceGroup()
+
+	instanceGroup.SetState(v1alpha1.ReconcileInit)
+	ctx.processParameters()
+
+	return ctx
+}
+
+func (ctx *InstanceGroupContext) processParameters() {
+	var (
+		instanceGroup = ctx.GetInstanceGroup()
+		spec          = instanceGroup.GetEKSFargateSpec()
+		params        = make(map[string]interface{})
+	)
+
+	params["ClusterName"] = spec.GetClusterName()
+	params["ProfileName"] = spec.GetProfileName()
+	params["Selectors"] = CreateFargateSelectors(spec.GetSelectors())
+	params["Subnets"] = CreateFargateSubnets(spec.GetSubnets())
+	params["Tags"] = CreateFargateTags(spec.GetTags())
+	ctx.AwsWorker.Parameters = params
+}
+
+func CreateFargateSubnets(subnets []string) []*string {
+	stringReferences := []*string{}
+	for _, s := range subnets {
+		temp := new(string)
+		*temp = s
+		stringReferences = append(stringReferences, temp)
+	}
+	return stringReferences
+}
+func CreateFargateTags(tagArray []map[string]string) map[string]*string {
+	tags := make(map[string]*string)
+	for _, t := range tagArray {
+		for k, v := range t {
+			va := new(string)
+			*va = v
+			tags[k] = va
+		}
+	}
+	return tags
+}
+
+// Convienence function to convert from json to API.
+func CreateFargateSelectors(selectors []v1alpha1.EKSFargateSelectors) []*eks.FargateProfileSelector {
+	var eksSelectors []*eks.FargateProfileSelector
+	for _, selector := range selectors {
+		m := make(map[string]*string)
+		for k, v := range selector.Labels {
+			vv := new(string)
+			*vv = v
+			m[k] = vv
+		}
+		eksSelectors = append(eksSelectors, &eks.FargateProfileSelector{Namespace: &selector.Namespace, Labels: m})
+	}
+	return eksSelectors
+}
+func IsRetryable(instanceGroup *v1alpha1.InstanceGroup) bool {
+	for _, state := range NonRetryableStates {
+		if state == instanceGroup.GetState() {
+			return false
+		}
+	}
+	return true
+}
+func (ctx *InstanceGroupContext) Create() error {
+	var arn string
+	instanceGroup := ctx.GetInstanceGroup()
+	spec := instanceGroup.GetEKSFargateSpec()
+	if instanceGroup.GetEKSFargateSpec().GetPodExecutionRoleArn() == "" {
+		created, err := ctx.AwsWorker.CreateDefaultRole()
+		if created || err != nil {
+			ctx.Log.Info("Creating default role")
+			return err
+		}
+		role, err := ctx.AwsWorker.GetDefaultRole()
+		if err != nil {
+			ctx.Log.Info("Failed to get default role", "error", err)
+			return err
+		}
+		arn = *role.Arn
+
+		err = ctx.AwsWorker.AttachDefaultPolicyToDefaultRole()
+		if err != nil {
+			ctx.Log.Info("Failed to get attach policy to role", "error", err)
+			return err
+		}
+		ctx.Log.Info("Attached default policy to role")
+
+	} else {
+		arn = spec.GetPodExecutionRoleArn()
+	}
+
+	ctx.Log.Info("Creating a profile.", "arn", arn)
```

Comment:
can you add to all log entries the name of the instancegroup:
e.g.
```
ctx.Log.Info("Creating a profile.", "instancegroup", instanceGroup.GetName(), "arn", arn)
```
That makes debugging a bit easier when there are many

Reply:
will do.